ostree/gen-fixtures.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+DIR=""
+TREE=$(mktemp -d /tmp/tree.XXXXX)
+USAGE="
+Usage: gen-fixtures.sh <output-dir>
+
+Generate an OSTree repository with (2) branches:
+ - rawhide
+ - stable
+
+Each branch will have (3) commits for versions:
+ - 1.0
+ - 1.1
+ - 1.2
+
+The OSTree repository will be created in <output-dir>.
+The directory will be be created.
+"
+
+show_help()
+{
+  echo "${USAGE}"
+}
+
+_mkdir()
+{
+  if [ -e ${DIR} ]; then
+    if [ ! -z "$(ls -A ${DIR})" ]; then
+      echo 1>&2 ""
+      echo 1>&2 "${DIR} must be empty."
+      echo 1>&2 ""
+      exit 1
+    fi
+  else
+    mkdir -p "${DIR}"
+  fi
+}
+
+gen_tree()
```

Comment:
> IRC, older versions of ostree would fail if there was not at least one file.

That explains a lot. Thank you.

I would recommend avoiding references to `$0`, and doing something else, like calling `touch "${TREE}/foo"`. There's [issues](http://mywiki.wooledge.org/BashFAQ/028) with `$0`.

> I also noticed that if the script exists because one of the two checks fails, the /tmp/tree.XX was not being cleaned up.

The following idiom is used in several of the pulp-fixtures scripts to ensure clean-up occurs in most situations:

```bash
# Create a workspace, and schedule it for deletion.
cleanup() { if [ -n "${working_dir:-}" ]; then rm -rf "${working_dir}"; fi }
trap cleanup EXIT  # bash pseudo signal
trap 'cleanup ; trap - SIGINT ; kill -s SIGINT $$' SIGINT
trap 'cleanup ; trap - SIGTERM ; kill -s SIGTERM $$' SIGTERM
working_dir="$(mktemp --directory)"
```

This basically says "if the script finishes executing, or if the script is nicely asked to die, then call `cleanup()`. AFAIK, Bash doesn't support exceptions, but this is pretty close.